server/stycoon/game.go

```diff
@@ -2,20 +2,250 @@ package stycoon
+		if err != nil {
+			return planets, fmt.Errorf("row read failed %v", err)
+		}
+		planet.Position = pos
+		planet.PrevPosition = posPrev
+		planet.Resources = map[string]TradingResource{}
+		err = game.setTradingResources(id, &planet.Resources)
+		if err != nil {
+			return planets, err
+		}
+		err = game.setPlanetResourcePrices(id, &planet.Resources)
+		if err != nil {
+			return planets, err
+		}
+
+		planets[strconv.Itoa(id)] = planet
+	}
+	if err = rows.Err(); err != nil {
+		return planets, fmt.Errorf("rows read failed: %v", err)
+	}
+	return planets, nil
+}
+
+func (game *Game) GetShips(playerId *int64) (map[string]ShipsValue, error) {
+	var ships = make(map[string]ShipsValue)
+	rows, err := game.db.Query("select o.`id`, o.`name`, o.`pos_x`, o.`pos_y`, o.`pos_x_prev`, o.`pos_y_prev`, s.`class`, o.`owner`, s.`life` from t_object as o join t_ship s on o.id = s.id")
+	if err != nil {
+		return ships, fmt.Errorf("query failed %v", err)
+	}
+	var id int
+	var ship ShipsValue
+	var pos = make([]int64, 2)
+	var posPrev = make([]int64, 2)
+	for rows.Next() {
+		err = rows.Scan(&id, &ship.Name, &pos[0], &pos[1], &posPrev[0], &posPrev[1], &ship.ShipClass, &ship.Player, &ship.Life)
+		if err != nil {
+			return ships, fmt.Errorf("row read failed %v", err)
+		}
+		ship.Position = pos
+		ship.PrevPosition = posPrev
+		ship.Resources = map[string]Resource{}
+		err = game.setResources(id, &ship.Resources)
+		if err != nil {
+			return ships, err
+		}
+		if playerId != nil && *playerId == ship.Player {
+			ship.Command, err = game.getCommand(id)
+			if err != nil {
+				return ships, err
+			}
+		}
+		ships[strconv.Itoa(id)] = ship
+	}
+	if err = rows.Err(); err != nil {
+		return ships, fmt.Errorf("rows read failed: %v", err)
+	}
+	return ships, nil
+}
+
+func (game *Game) getCommand(shipId int) (Command, error) {
+	var command Command
+	err := game.db.QueryRow("select `type`, `target`, `resource`, `amount`, `class` from t_command where ship = ?", shipId).Scan(&command.Type, &command.Target, &command.Resource, &command.Amount, &command.ShipClass)
+	if err != nil {
+		return command, err
+	}
+	return command, nil
+}
+
+func (game *Game) setPlanetResourcePrices(planetId int, planetResources *map[string]TradingResource) error {
+	rows, err := game.db.Query("select `resource`, `buy`, `sell`  from t_price where planet = ?", planetId)
+	if err != nil {
+		return fmt.Errorf("query failed %v", err)
+	}
+	var resourceId int
+	var buy, sell sql.NullFloat64
+	for rows.Next() {
+		err = rows.Scan(&resourceId, &buy, &sell)
+		if err != nil {
+			return fmt.Errorf("row read failed %v", err)
+		}
+		if buy.Valid {
+			resource, ok := (*planetResources)[strconv.Itoa(resourceId)]
+			if ok {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{Amount: resource.Amount, BuyPrice: buy.Float64}
+			} else {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{BuyPrice: buy.Float64}
```

Comment:
should not just amount be set to 0 in this case?

Reply:
that is default value